web/src/PreviewList.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { PureComponent } from "react"
+import PathBuilder from "./PathBuilder"
+import { Link } from "react-router-dom"
+import "./PreviewList.scss"
+
+type PreviewListProps = {
+  pathBuilder: PathBuilder
+  resourcesWithEndpoints: Array<string>
+}
+
+class PreviewList extends PureComponent<PreviewListProps> {
+  render() {
+    let pb = this.props.pathBuilder
+    if (this.props.resourcesWithEndpoints.length === 0) {
+      return <p>No resources found with endpoints!</p>
+    }
+    let links = this.props.resourcesWithEndpoints.map((r, i) => (
+      <li key={r + i}>
```

Comment:
shouldn't the key just be `r`?

Reply:
I was future proofing a little bit here and imagining a future where there are multiple endpoints for the same name. But then the list wouldn't make sense. So yeah I'll just delete that.